clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py

```diff
@@ -1,37 +1,107 @@
-from clarity_ext.domain.validation import UsageError
-from clarity_ext_scripts.covid.rtpcr_analyse_script import RT_PCR_NEGATIVE_CONTROL, RT_PCR_POSITIVE_CONTROL
+
+import logging
+
+
+# TODO Move these constants into some other file
+from clarity_ext_scripts.covid.partner_api_client import VALID_COVID_RESPONSES, \
+    COVID_RESPONSE_FAILED, COVID_RESPONSE_NEGATIVE, COVID_RESPONSE_POSITIVE
+
+
+log = logging.getLogger(__name__)
+
+
+class AnalysisServiceException(Exception):
+    pass
+
+
+class MultipleAnalysisErrors(AnalysisServiceException):
+    pass
+
+
+class PositiveControlWasNegative(AnalysisServiceException):
+    pass
+
+
+class NegativeControlWasPositive(AnalysisServiceException):
+    pass
 
 
 class RTPCRAnalysisService(object):
-    """
-    A service to analyze if a sample is positive for Covid19
-    Given:
-    CT value of sample
-    Discriminator for assay (lower bound, upper bound)
-    It's expected to be 2 controls on a plate, and these are to be
-    validated before ordinary samples are analyzed.
-    """
+
+    HEX_THRESHOLD = 32
+    FAM_THRESHOLD = 38
+
     def __init__(self):
-        self.has_valid_pos_control = False
-        self.has_valid_neg_control = False
+        pass
+
+# Understanding of the analysis criterias
+# FAM = COVID, HEX = Human internal control
+# Positive if CT(FAM) = <=38
+# Failed  if CT(FAM)= >38
+# Negative if CT(FAM) = 0 and CT(HEX) = <=32
+# Retest if CT(FAM) = 0 and CT(HEX) = >32
+
+# TODO Should we issue a re-test status here
+# TODO Later we might have to account for giving two trails over 38 as positive.
+
+    def _analyze_sample(self, sample):
+        fam_ct = sample["FAM-CT"]
+        hex_ct = sample["HEX-CT"]
+        if fam_ct > self.FAM_THRESHOLD:
+            return COVID_RESPONSE_FAILED
+        elif fam_ct == 0 and hex_ct <= self.HEX_THRESHOLD:
+            return COVID_RESPONSE_NEGATIVE
+        elif fam_ct == 0 and hex_ct > self.HEX_THRESHOLD:
+            # TODO Should retest and failed be the same thing or not.
```

Comment:
I think it's better that we give different results here, and we then interpret "retest" in the extension as a failure. Then it's gonna be quite straightforward change when we implement retest.

It's also good that we set that constant on the sample level. It will allow the lab to see how many should have been re-tested, if we implemented it.

Reply:
Probably add RESTEST Status here, check with Maike.